tiledb/sm/filesystem/posix.cc

```diff
@@ -311,7 +311,7 @@ tuple<Status, optional<std::vector<directory_entry>>> Posix::ls_with_sizes(
   struct dirent* next_path = nullptr;
   DIR* dir = opendir(path.c_str());
   if (dir == nullptr) {
-    return {Status::Ok(), nullopt};
+    return {Status::Ok(), std::vector<directory_entry{}};
```

Comment:
Should we check `errno` and return an empty vector only if it is `ENOENT`, and fail otherwise?

Reply:
I'm only narrowly fixing a specific defect. If you want to make other improvements, feel free to in another PR. Any further changes are beyond the scope of this one-line PR.